core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -368,7 +368,13 @@ private void renameToFinal(FileSystem fs, Path src, Path dst, int nextVersion) {
       }
 
       if (fs.exists(dst)) {
-        throw new CommitFailedException("Version %d already exists: %s", nextVersion, dst);
+        CommitFailedException cfe =
+            new CommitFailedException("Version %d already exists: %s", nextVersion, dst);
+        RuntimeException re = tryDelete(src);
```

Comment:
is it always correct to delete the src file or could there be cases where we'd want to keep it?

Reply:
@nastra  yes, i think it is always safe to delete the temp src file(temp metadata file), the commit retry would create a new temp metadata file.

Reply:
@nastra would you please help to take another look? thanks